### PR TITLE
Migration of htmlAudioPlayer to ES6 module

### PR DESCRIPTION
**Pull request by Camc314**
**Issues**
part of #868

### CODE REVIEW

File: src/plugins/htmlAudioPlayer/plugin.js [hunk 1]
```diff
@@ -391,124 +394,123 @@ define(['events', 'browser', 'require', 'apphost', 'appSettings', 'htmlMediaHelp
...
-        var list = [];
-        var audio = document.createElement('audio');
-
-        if (typeof audio.playbackRate === 'number') {
-            list.push('PlaybackRate');
-        }
+    }
 
-        return list;
+    destroy() {
```
Comment (MrTimscampi) [inline src/plugins/htmlAudioPlayer/plugin.js:491]: No need for empty functions.

Comment (Camc314) [reply src/plugins/htmlAudioPlayer/plugin.js]: Removed 👍

File: src/plugins/htmlAudioPlayer/plugin.js [hunk 2]
```diff
@@ -1,91 +1,94 @@
...
+import browser from 'browser';
+import appHost from 'apphost';
+import * as htmlMediaHelper from 'htmlMediaHelper';
+
+function getDefaultProfile() {
+    return new Promise(function (resolve, reject) {
+        import('browserdeviceprofile').then(({default: profileBuilder}) => {
+            resolve(profileBuilder({}));
         });
+    });
```
Comment (dmitrylyzo) [inline src/plugins/htmlAudioPlayer/plugin.js:11]: I believe we could remove top `Promise` and do everything with `import`. Leave it for "Promise Cleanup" PR?

File: src/plugins/htmlAudioPlayer/plugin.js [hunk 3]
```diff
@@ -1,91 +1,94 @@
...
-                    if (contentType === 'application/x-mpegurl') {
-                        resolve();
-                    } else {
-                        reject();
-                    }
-                }, reject);
-            });
+    // issue head request to get content type
+    return new Promise(function (resolve, reject) {
+        import('fetchHelper').then(({default: fetchHelper}) => {
```
Comment (dmitrylyzo) [inline src/plugins/htmlAudioPlayer/plugin.js:73]: `fetchHelper` doesn't have `default` export.
```suggestion
        import('fetchHelper').then((fetchHelper) => {
```
Or we must add `default` export.

File: src/plugins/htmlAudioPlayer/plugin.js (hunk 2 above)
Comment (Camc314) [reply src/plugins/htmlAudioPlayer/plugin.js]: I've removed the promise

File: src/plugins/htmlAudioPlayer/plugin.js (hunk 3 above)
Comment (Camc314) [reply src/plugins/htmlAudioPlayer/plugin.js]: Fixed

**Pull request merged**